src/reynir_correct/pattern.py

```diff
@@ -917,6 +920,226 @@ def vera_að(self, match: SimpleTree) -> None:
             )
         )
 
+    def dir_loc(self, match: SimpleTree) -> None:
+        adv = match.first_match("( 'inn'|'út'|'upp' )")
+        pp = match.first_match("PP > { P > { ( 'í'|'á'|'um' ) } NP > { ( no_et_þgf|no_ft_þgf|pfn_et_þgf|pfn_ft_þgf ) } }")  
+        if pp is None:
+            pp = match.first_match("PP > { P > { ( 'í'|'á'|'um' ) } NP > { ( no_et_þf|no_ft_þf|pfn_et_þf|pfn_ft_þf ) } }") 
+        assert adv is not None
+        assert pp is not None
+        start, end = min(adv.span[0], pp.span[0]), max(adv.span[1], pp.span[1])
+        if adv.span < pp.span:
+            if pp.tidy_text.startswith(adv.tidy_text):
+                narrow_match = pp.tidy_text
+            else:
+                narrow_match = adv.tidy_text + ' ' + pp.tidy_text
+        elif pp.span < adv.span:
+            narrow_match = pp.tidy_text + ' ' + adv.tidy_text
+        correction = adv.tidy_text+'i'
+        text = "Hér á líklega að vera {0} í stað {1}".format(
+            correction, adv.tidy_text
+        )
+        detail = "Í samhenginu '{0}' er rétt að nota atviksorðið '{1}' í stað '{2}'.".format(
+            narrow_match, correction, adv.tidy_text
```

Comment:
Pylance complains that `narrow_match `is possibly unbound - it is not guaranteed to have received a value from the if statements above.